.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1262,42 +1262,57 @@ private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject refe
             var ret = new DateTimeResolutionResult();
 
             var er = this.config.DateExtractor.Extract(text, referenceDate);
+            var pr1 = new DateTimeParseResult();
+            var pr2 = new DateTimeParseResult();
             if (er.Count < 2)
             {
                 er = this.config.DateExtractor.Extract(this.config.TokenBeforeDate + text, referenceDate);
-                if (er.Count < 2)
-                {
-                    return ret;
+                if (er.Count >= 2)
+                {
+                    er[0].Start -= this.config.TokenBeforeDate.Length;
+                    er[1].Start -= this.config.TokenBeforeDate.Length;
+                }
+                else
+                {
+                    var nowPr = ParseNowAsDate(text, referenceDate);
+                    if (nowPr.Value == null || er.Count < 1)
+                    {
+                        return ret;
+                    }
+
+                    var datePr = this.config.DateParser.Parse(er[0], referenceDate);
+                    pr1 = datePr.Start < nowPr.Start ? datePr : nowPr;
+                    pr2 = datePr.Start < nowPr.Start ? nowPr : datePr;
                 }
-
-                er[0].Start -= this.config.TokenBeforeDate.Length;
-                er[1].Start -= this.config.TokenBeforeDate.Length;
             }
 
-            var match = this.config.WeekWithWeekDayRangeRegex.Match(text);
-            string weekPrefix = null;
-            if (match.Success)
-            {
-                weekPrefix = match.Groups["week"].ToString();
-            }
+            if (er.Count >= 2)

```

Comment:
Why do we need this if now?

Reply:
In the former case ”er.count<2“, the value of er.count maybe changed in

er = this.config.DateExtractor.Extract(this.config.TokenBeforeDate + text, referenceDate);

so, i  judge the value again. if the value>=2, we should parse the er to pr, otherwise not needed